File: src/CSharp/MicrosoftEdgeLegacyReborn/Converters.cs
using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;

namespace MicrosoftEdgeLegacyReborn
{
    /// <summary>
    /// Converts a Boolean value to a Visibility value.
    /// </summary>
    public class BooleanToVisibilityConverter : IValueConverter
    {
        /// <summary>
        /// Converts a <see cref="bool"/> to a <see cref="Visibility"/>.
        /// </summary>
        /// <param name="value">The boolean value to convert.</param>
        /// <param name="targetType">The target type of the conversion.</param>
        /// <param name="parameter">An optional parameter.</param>
        /// <param name="language">The language of the conversion.</param>
        /// <returns>
        /// Returns <see cref="Visibility.Visible"/> if <see langword="true"/>, otherwise returns <see cref="Visibility.Collapsed"/>.
        /// </returns>
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if ((bool)value)
            {
                return Visibility.Visible;
            }
            else
            {
                return Visibility.Collapsed;
            }
        }

        /// <summary>
        /// Converts back a <see cref="Visibility"/> value to a <see cref="bool"/>.
        /// </summary>
        /// <param name="value">The visibility value to convert.</param>
        /// <param name="targetType">The target type of the conversion.</param>
        /// <param name="parameter">An optional parameter used to indicate if the conversion should be reversed.</param>
        /// <param name="language">The language of the conversion.</param>
        /// <returns>
        /// Returns <see langword="true"/> if the visibility is <see cref="Visibility.Visible"/>; otherwise, returns <see langword="false"/>.
        /// If the parameter is "Reverse", the logic is inverted.
        /// </returns>
        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return (Visibility)value == Visibility.Visible ^ (parameter as string ?? string.Empty).Equals("Reverse");
        }
    }

    public class BooleanToStringSelectorConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return (bool)value ? ((string)parameter).Split("|")[0] : (object)((string)parameter).Split("|")[1];
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }

    public class BooleanToColorSelecter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return new SolidColorBrush(SharedFunctions.GetColorFromHex((bool)value ? ((string)parameter).Split("|")[0] : ((string)parameter).Split("|")[1]));
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}

File: src/Cpp/MicrosoftEdgeLegacyReborn/Converters.cpp
#include "pch.h"
#include "Converters.h"
using namespace std;
using namespace MicrosoftEdgeLegacyReborn;

Object^ BooleanToVisibilityConverter::Convert(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    bool boolValue = safe_cast<bool>(value);
    return boolValue ? Visibility::Visible : Visibility::Collapsed;
}

Object^ BooleanToVisibilityConverter::ConvertBack(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    Visibility visibilityValue = safe_cast<Visibility>(value);
    String^ param = parameter ? parameter->ToString() : "";
    return (visibilityValue == Visibility::Visible) != (param == "Reverse");
}

Object^ BooleanToStringSelectorConverter::Convert(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    wstring wParam(safe_cast<String^>(parameter)->Data());

    // Dividimos el parámetro usando '|' como delimitador
    size_t delimiterPos = wParam.find(L'|');
    if (delimiterPos == wstring::npos)
    {
        throw ref new InvalidArgumentException("El parámetro no contiene el delimitador '|'.");
    }

    wstring part1 = wParam.substr(0, delimiterPos);
    wstring part2 = wParam.substr(delimiterPos + 1);

    return ref new String((safe_cast<bool>(value) ? part1 : part2).c_str());
}

Object^ BooleanToStringSelectorConverter::ConvertBack(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    throw ref new NotImplementedException();
}

Object^ BooleanToColorSelectorConverter::Convert(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    bool boolValue = safe_cast<bool>(value);
    String^ paramStr = safe_cast<String^>(parameter);
    wstring wParam(paramStr->Data());

    // Dividimos el parámetro usando '|' como delimitador
    size_t delimiterPos = wParam.find(L'|');
    if (delimiterPos == wstring::npos)
    {
        throw ref new InvalidArgumentException("El parámetro no contiene el delimitador '|'.");
    }

    wstring color1 = wParam.substr(0, delimiterPos);
    wstring color2 = wParam.substr(delimiterPos + 1);

    // Convertimos a color y creamos el brush
    String^ colorHex = ref new String((boolValue ? color1 : color2).c_str());
    return ref new SolidColorBrush(SharedFunctions::GetColorFromHex(colorHex));
}

Object^ BooleanToColorSelectorConverter::ConvertBack(Object^ value, TypeName targetType, Object^ parameter, String^ language)
{
    throw ref new NotImplementedException();
}